meta-mel/recipes-support/consolekit/consolekit_%.bbappend

```diff
@@ -1,5 +1,36 @@
 FILESEXTRAPATHS_prepend := "${THISDIR}/${PN}:"
 
 SRC_URI_append = " \
-    file://0001-Ensure-filesystems-are-still-mounted-when-consolekit.patch \
+    file://console-kit-log-system-shutdown.service.in \
+    file://console-kit-shutdown-handler \
 "
+
+RDEPENDS_${PN} += "bash"
+
+do_install_append() {
+
+    shutdown_handler="console-kit-shutdown-handler"
+    shutdown_service="console-kit-log-system-shutdown.service"
+
+    # install script with logic to choose between "stop" and "restart"
+    install -d ${D}${sbindir}
+    install -m 755 ${WORKDIR}/${shutdown_handler} ${D}${sbindir}
+
+    # install "shutdown" service and auto-start softlink
+    sed -e 's,@sbindir@,${sbindir},g' ${WORKDIR}/${shutdown_service}.in > ${WORKDIR}/${shutdown_service}
+    install -d ${D}${systemd_system_unitdir}
+    install -m 644 ${WORKDIR}/${shutdown_service} ${D}${systemd_system_unitdir}
+
+    install -d ${D}${systemd_system_unitdir}/basic.target.wants
+    pushd ${D}${systemd_system_unitdir}/basic.target.wants
+    ln -s ../${shutdown_service}
+    popd
```

Comment:
Rather than using pushd and popd I believe this can be handled in a one liner where the ln command would change to
`ln -sf ../${shutdown_service} ${D}${systemd_system_unitdir}/basic.target.wants/`